typing/ctype.ml

```diff
@@ -1620,7 +1627,8 @@ let expand_abbrev_gen kind find_type_expansion env ty =
         (* prerr_endline
            ("found a "^string_of_kind kind^" expansion for "^Path.name path);*)
         if level <> generic_level then update_level env level ty';
-        update_scope scope ty';
+        if not !Clflags.principal then update_scope scope ty'
+        else if get_scope ty' <> scope then raise_scope_escape_exn ty;
```

Comment:
I would find it simpler to just call `update_scope` here unconditionally, as before; it should be equivalent, right?

Reply:
After a last look, always updating is probably fine, but it is only required if we actually use the type, which is not the case in principal mode anymore.
There might be some strange behavior due to the non-uniformity of the expansion; again I do not see how to fix that without keep-expansion.